packages/nextra-theme-docs/src/components/highlight-matches.tsx

```diff
@@ -15,27 +41,27 @@ export const HighlightMatches = memo<MatchArgs>(function HighlightMatches({
     return null
   }
   const splitText = value.split('')
-  const escapedSearch = escapeStringRegexp(match.trim())
-  const regexp = new RegExp(escapedSearch.replaceAll(' ', '|'), 'ig')
```

Comment:
thank you for reporting the new issue with the memory leak, I am wondering what would happen if we just replace multiple spaces in this place?
```js
const regexp = new RegExp(escapedSearch.replaceAll(/\s+/g, '|'), 'ig')
```

Reply:
Thank you for your feedback!

Given that `highlight-matches.tsx` is working under the assumption that a search term has been entered, replacing multiple whitespaces is indeed sufficient to address the memory leak issue.

I've updated the PR to reflect this change.